anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/mapper/DetectorMapperCache.java

```diff
@@ -117,59 +116,59 @@ private MetricSet metricsFor(String cacheName, final Cache cache) {
      * </pre>
      *
      * @param disabledMappings the list of mappings
      */
     public void removeDisabledDetectorMappings(List<DetectorMapping> disabledMappings) {
-        List<UUID> detectorIdsOfDisabledMappings = disabledMappings.stream().map(detectorMapping -> detectorMapping.getDetector().getUuid()).collect(Collectors.toList());
+        List<Detector> detectorsOfDisabledMappings = disabledMappings.stream()
+                .map(detectorMapping -> detectorMapping.getDetector())
+                .collect(Collectors.toList());
 
         Map<String, String> mappingsWhichNeedsAnUpdate = new HashMap<>();
 
-        this.cache.asMap().forEach((key, bunchOfDetectorIds) -> {
-            if (detectorIdsOfDisabledMappings.stream().anyMatch(uuid -> bunchOfDetectorIds.contains(uuid.toString()))) {
-                mappingsWhichNeedsAnUpdate.put(key, bunchOfDetectorIds);
+        this.cache.asMap().forEach((key, value) -> {
+            if (detectorsOfDisabledMappings.stream().anyMatch(detector -> value.contains(detector.getUuid().toString()))) {
+                mappingsWhichNeedsAnUpdate.put(key, value);
             }
         });
 
-
         Map<String, String> modifiedDetectorMappings = new HashMap<>();
-        mappingsWhichNeedsAnUpdate.forEach((key, bunchOfDetectorIds) -> {
-            String bunchOfUpdatedDetectorIds =
-                    removeDisabledDetectorIds(detectorIdsOfDisabledMappings, bunchOfDetectorIds);
-            modifiedDetectorMappings.put(key, bunchOfUpdatedDetectorIds);
+
+        mappingsWhichNeedsAnUpdate.forEach((key, value) -> {
+            String newValue = removeDisabledDetectors(detectorsOfDisabledMappings, value);
+            modifiedDetectorMappings.put(key, newValue);
         });
 
         log.info("removing mappings : {} from cache entries",
-                Arrays.toString(detectorIdsOfDisabledMappings.toArray()));
+                Arrays.toString(detectorsOfDisabledMappings.toArray()));
         modifiedDetectorMappings.forEach((key, value) -> log.info("cache key: {}, updated mapping {}", key, value));
 
         this.cache.putAll(modifiedDetectorMappings);
     }
 
-    private String removeDisabledDetectorIds(List<UUID> detectorUuids, String detectorIdsString) {
-        List<Detector> detectors = CacheUtil.buildDetectors(detectorIdsString);
-        detectorUuids.forEach(uuid -> {
-            detectors.remove(new Detector(uuid));
+    private String removeDisabledDetectors(List<Detector> detectorsToBeDisabled, String value) {
+        List<Detector> detectorsStoredInCache = CacheUtil.buildDetectors(value);
+        detectorsToBeDisabled.forEach(detectorToBeDisabled -> {
+            detectorsStoredInCache.remove(new Detector(detectorToBeDisabled.getConsumerId(), detectorToBeDisabled.getUuid()));
         });
-        return CacheUtil.getDetectorIds(detectors);
+        return CacheUtil.getValue(detectorsStoredInCache);
```

Comment:
Similar comments as I made above - `getValue()` is much less clear than `getDetectorIds()`. Please reconsider the names here.